src/main/java/Quag.java

```diff
@@ -0,0 +1,150 @@
+            switch (command) {
+                case "quag":
+                    exitProgram();
+                    return;
+                case "list":
+                    displayList();
+                    break;
+                case "mark":
+                    markTask(userInput);
+                    break;
+                case "unmark":
+                    unmarkTask(userInput);
+                    break;
+                case "todo":
+                    addTodo(userInput);
+                    break;
+                case "deadline":
+                    addDeadline(userInput);
+                    break;
+                case "event":
+                    addEvent(userInput);
+                    break;
+                default:
+                    displayCommandList();
+                    break;
+            }
+        }
+    }
+
+    private static String getCommand(String userInput) {
+        //find first space
+        int firstSpaceIndex = userInput.indexOf(" ");
+        // first word is the command
+        if (firstSpaceIndex != -1) {
+            return userInput.substring(0, firstSpaceIndex).toLowerCase();
+        }
+        // entire userInput is the command
+        else {
+            return userInput.toLowerCase();
+        }
+    }
+
+
+    private static void exitProgram() {
+        System.out.println("_______________________________________");
+        System.out.println("See you soon! quag quag");
+        System.out.println("_______________________________________");
+    }
+
+    private static void displayList() {
+        ListUtils.displayList();
+    }
+
+
+    private static void markTask(String userInput) {
+        String[] parts = userInput.split(" ");
+        if (parts.length >= 2) {
+            try {
+                int taskIndex = Integer.parseInt(parts[1]);
+                ListUtils.markAsDone(taskIndex);
+            } catch (NumberFormatException e) {
+                System.out.println("That's not a quaggin number!!");
+            }
+        } else {
+            System.out.println("There's no quaggin number!!");
+        }
+    }
+    private static void unmarkTask(String userInput) {
+        String[] parts = userInput.split(" ");
+        if (parts.length >= 2) {
+            try {
+                int taskIndex = Integer.parseInt(parts[1]);
+                ListUtils.markAsNotDone(taskIndex);
+            } catch (NumberFormatException e) {
+                System.out.println("That's not a quaggin number!!");
+            }
+        } else {
+            System.out.println("There's no quaggin number!!");
+        }
+    }
+
+    private static void addTodo(String userInput) {
+        //description is from parts[1] to end
+        String description = userInput.substring("todo".length()).trim();
+        ListUtils.addTodoToList(description);
+    }
+
+    private static void addDeadline(String userInput) {
+        //description is from parts[1] until / is detected
```

Comment:
Code should be as modular as possible, and global variables should be used minimally (including references in the comments). It would be difficult to always trace to a previous function to understand what parts [1] means.